pkg/suggestion/grid_service_test.go

```diff
@@ -0,0 +1,85 @@
+package suggestion
+
+import (
+	"testing"
+)
+
+func TestAllocInt(t *testing.T) {
```

Comment:
Could you add test for `genGrids`, `parseSuggestParam `, and `setP`?

Reply:
ok. Please wait for a while.

Reply:
GenGrid, parseSuggestParam, and setP test added.